src/native/fs/fs_napi.cpp

```diff
@@ -213,6 +224,258 @@ struct Stat : public FSWorker
     }
 };
 
+/**
+ * Getxattr is an fs op
+ */
+struct Getxattr : public FSWorker
+{
+    std::string _path;
+    std::vector<KeyValue> _md;
+    Getxattr(const Napi::CallbackInfo& info) 
+        : FSWorker(info)
+        , _md(0)
+    {
+        _path = info[1].As<Napi::String>();
+        Begin(XSTR() << "Getxattr " << DVAL(_path));
+    }
+    virtual void Work()
+    {
+        ssize_t buf_len, key_len, value_len;
+        char *buf, *key, *val;
```

Comment:
This is unsafe c style. 
Best move closed to scope that uses these to avoid dangling pointers.